packages/backend-core/src/features/index.ts

```diff
@@ -1,79 +1,108 @@
-  const flags = getTenantFeatureFlags(tenantId)
-  return flags.includes(featureFlag)
+// Exported for use in tests, should not be used outside of this file.
+export function defaultFlags(): Flags {
+  return cloneDeep(DEFAULTS)
 }
 
-export function getTenantFeatureFlags(tenantId: string) {
-  let flags: string[] = []
-  const envFlags = buildFeatureFlags()
-  if (envFlags) {
-    const globalFlags = envFlags["*"]
-    const tenantFlags = envFlags[tenantId] || []
-
-    // Explicitly exclude tenants from global features if required.
-    // Prefix the tenant flag with '!'
-    const tenantOverrides = tenantFlags.reduce(
-      (acc: string[], flag: string) => {
-        if (flag.startsWith("!")) {
-          let stripped = flag.substring(1)
-          acc.push(stripped)
-        }
-        return acc
-      },
-      []
-    )
-
-    if (globalFlags) {
-      flags.push(...globalFlags)
-    }
-    if (tenantFlags.length) {
-      flags.push(...tenantFlags)
+function isFlagName(name: string): name is keyof Flags {
+  return FLAGS[name as keyof typeof FLAGS] !== undefined
+}
+
+/**
+ * Reads the TENANT_FEATURE_FLAGS environment variable and returns a Flags object
+ * populated with the flags for the current tenant, filling in the default values
+ * if the flag is not set.
+ *
+ * Check the tests for examples of how TENANT_FEATURE_FLAGS should be formatted.
+ *
+ * In future we plan to add more ways of setting feature flags, e.g. PostHog, and
+ * they will be accessed through this function as well.
+ */
+export async function fetch(): Promise<Flags> {
+  const currentTenantId = context.getTenantId()
+  const flags = defaultFlags()
+
+  const split = (env.TENANT_FEATURE_FLAGS || "")
+    .split(",")
+    .map(x => x.split(":"))
+  for (const [tenantId, ...features] of split) {
+    if (!tenantId || (tenantId !== "*" && tenantId !== currentTenantId)) {
+      continue
     }
 
-    // Purge any tenant specific overrides
-    flags = flags.filter(flag => {
-      return tenantOverrides.indexOf(flag) == -1 && !flag.startsWith("!")
-    })
+    for (let feature of features) {
+      let value = true
+      if (feature.startsWith("!")) {
+        feature = feature.slice(1)
+        value = false
+      }
+
+      if (!isFlagName(feature)) {
+        throw new Error(`Feature: ${feature} is not an allowed option`)
+      }
+
+      if (typeof flags[feature] !== "boolean") {
+        throw new Error(`Feature: ${feature} is not a boolean`)
+      }
+
+      // @ts-ignore
+      flags[feature] = value
+    }
   }
 
   return flags
 }
 
-export enum TenantFeatureFlag {
-  LICENSING = "LICENSING",
-  GOOGLE_SHEETS = "GOOGLE_SHEETS",
-  USER_GROUPS = "USER_GROUPS",
-  ONBOARDING_TOUR = "ONBOARDING_TOUR",
+// Gets a single feature flag value. This is a convenience function for
+// `fetch().then(flags => flags[name])`.
+export async function get<K extends keyof Flags>(name: K): Promise<Flags[K]> {
+  const flags = await fetch()
```

Comment:
In a follow-up PR when I introduce PostHog, I suspect we will need to start caching feature flags per-tenant to avoid excessively calling out to the PostHod `decide` API.

Reply:
That's very odd - perchance is this just the language server in VSCode having a hard time? I've checkout this branch and removed the `// ts-ignore` - with a build and a check:types I wasn't seeing any errors here.